subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/AbstractMavenPublishJavaIntegTest.groovy

```diff
@@ -277,12 +277,11 @@ abstract class AbstractMavenPublishJavaIntegTest extends AbstractMavenPublishInt
         javaLibrary.assertPublished()
 
         javaLibrary.parsedPom.packaging == null // 'jar' packaging
-        javaLibrary.parsedPom.scopes.keySet() == ["compile", "runtime"] as Set
+        javaLibrary.parsedPom.scopes.keySet() == ["compile", "no_scope", "runtime"] as Set
         javaLibrary.parsedPom.scopes.compile.assertDependsOn("org.springframework:spring-core:1.2.9")
-        javaLibrary.parsedPom.scopes.compile.assertDependencyManagement("commons-logging:commons-logging:1.1")
 
         javaLibrary.parsedPom.scopes.runtime.assertDependsOn("org.apache.commons:commons-compress:1.5")
-        javaLibrary.parsedPom.scopes.runtime.assertDependencyManagement("commons-logging:commons-logging:1.2", "org.tukaani:xz:1.6")
+        javaLibrary.parsedPom.scopes.no_scope.assertDependencyManagement("commons-logging:commons-logging:1.1", "commons-logging:commons-logging:1.2", "org.tukaani:xz:1.6")
```

Comment:
:o: mmmm, so we get commons-logging twice with different versions?

Reply:
Yes, the de-duplication system we have only works if the versions are the same in the declaration.
I did not attempt to improve that because it might be a complicated thing to get it perfectly right.
And in the end it matches what Gradle Module Metadata shows for `java-runtime`, even though Maven will only consider the last declaration in the file.